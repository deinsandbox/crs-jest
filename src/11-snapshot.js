export const getCharacter = (data) => ({
  id: data.id,
  name: data.name,
  status: data.status,
  species: data.species,
  gender: data.gender,
});

export const getUser = () => ({
  name: "equiman",
  createAt: new Date(),
  id: Math.floor(Math.random() * 20),
});
