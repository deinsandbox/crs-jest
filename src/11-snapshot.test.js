import { getCharacter, getUser } from "./11-snapshot";
import character from "./11-rick.json";

describe("snapshot.js", () => {
  test("should compare a snapshot", () => {
    const result = getCharacter(character);
    expect(result).toMatchSnapshot();
  });

  test.skip("alway should fail the snapshot", () => {
    const result = getUser();
    expect(result).toMatchSnapshot();
  });

  test("should use snapshot exceptions", () => {
    const result = getUser();

    const propertyMatchers = {
      createAt: expect.any(Date),
      id: expect.any(Number),
    };
    expect(result).toMatchSnapshot(propertyMatchers);
  });
});

// Run Test: npm test snapshot
// Update Snapshot: npm test snapshot -u
