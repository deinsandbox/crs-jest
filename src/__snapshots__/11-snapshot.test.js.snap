// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot.js alway should fail the snapshot 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "equiman",
}
`;

exports[`snapshot.js should compare a snapshot 1`] = `
Object {
  "gender": "male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "human",
  "status": "alive",
}
`;

exports[`snapshot.js should use snapshot exceptions 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
  "name": "equiman",
}
`;
